src/System.Management.Automation/engine/SessionState.cs

```diff
@@ -65,6 +66,9 @@ internal SessionStateInternal(SessionStateInternal parent, bool linkToGlobal, Ex
             // is used for the pushd and popd commands
 
             _workingLocationStack = new Dictionary<String, Stack<PathInfo>>(StringComparer.OrdinalIgnoreCase);
+            // conservative choice to limit the memory impact in case of a regression
+            const int locationHistoryLimit = 20;
+            _workingLocationHistoryStack = new BoundedStack<PathInfo>(locationHistoryLimit);
```

Comment:
This variable name is easily confused with _workingLocationStack and _workingLocationHistoryStack. I would suggest some more meaningful like _SetLocationHistory. (I don't think it needs it's type in it's name, and if we do decide to implement `cd +` it won't be a stack anymore.)

Reply:
@bergmeister Please replace `_SetLocationHistory` -> `_setLocationHistory` - it is our name convention for static variables.